src/discof/restore/utils/fd_ssarchive.c

```diff
@@ -140,3 +155,76 @@ fd_ssarchive_latest_pair( char const * directory,
   if( FD_UNLIKELY( errno && errno!=ENOENT ) ) FD_LOG_ERR(( "readdir() failed (%i-%s)", errno, fd_io_strerror( errno ) ));
   return 0;
 }
+
+void
+fd_ssarchive_remove_old_snapshots( char const * directory,
+                                   uint         max_full_snapshots_to_keep,
+                                   uint         max_incremental_snapshots_to_keep ) {
+  ulong full_snapshots_cnt        = 0UL;
+  ulong incremental_snapshots_cnt = 0UL;
+  fd_ssarchive_entry_t full_snapshots[ FD_SSARCHIVE_MAX_FULL_ENTRIES ];
+  fd_ssarchive_entry_t incremental_snapshots[ FD_SSARCHIVE_MAX_INCREMENTAL_ENTRIES ];
+
+  DIR * dir = opendir( directory );
+  if( FD_UNLIKELY( !dir ) ) {
+    if( FD_LIKELY( errno==ENOENT ) ) FD_LOG_ERR(( "opendir() failed `%s` (%i-%s)", directory, errno, fd_io_strerror( errno ) ));
```

Comment:
```suggestion
    if( FD_LIKELY( errno!=ENOENT ) ) FD_LOG_ERR(( "opendir() failed `%s` (%i-%s)", directory, errno, fd_io_strerror( errno ) ));
```

ENOENT is fine here, right?

Reply:
fixed